lib/fs.js

```diff
@@ -259,13 +259,13 @@ Stats.prototype.isSocket = function() {
 
 const statValues = binding.getStatValues();
 
-function statsFromValues() {
-  return new Stats(statValues[0], statValues[1], statValues[2], statValues[3],
-                   statValues[4], statValues[5],
-                   statValues[6] < 0 ? undefined : statValues[6], statValues[7],
-                   statValues[8], statValues[9] < 0 ? undefined : statValues[9],
-                   statValues[10], statValues[11], statValues[12],
-                   statValues[13]);
+function statsFromValues(values = statValues) {
```

Comment:
This change is necessary because using an async function for stat operations means we have a race condition on the use of `statValues` when there's just a single TypedArray. So, for the normal callback case, we keep the single shared typed array, but for the promises version, we allocate one TypedArray per Promise. It's a bit more expensive for the promises case, but keeps things status quo for the callback case.